modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/walkthrough/Walkthrough.js

```diff
@@ -0,0 +1,422 @@
+import {OverlayMask} from '@clayui/core';
+import {ClayCheckbox} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import ClayPopover from '@clayui/popover';
+import {ReactPortal, usePrevious} from '@liferay/frontend-js-react-web';
+import PropTypes from 'prop-types';
+import React, {
+	forwardRef,
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from 'react';
+
+import {doAlign} from './doAlign';
+import {useObserveRect} from './useObserveRect';
+
+const Hotspot = forwardRef(({onHotspotClick, trigger}, ref) => {
+	const align = useCallback(() => {
+		if (trigger && ref?.current) {
+			doAlign({
+				points: ['cc', 'tl'],
+				sourceElement: ref.current,
+				targetElement: trigger,
+			});
+		}
+	}, [ref, trigger]);
+
+	useEffect(() => {
+		align();
+	}, [align]);
+
+	useObserveRect(align, trigger);
+
+	return (
+		<div
+			className="lfr-walkthough-hotspot"
+			onClick={onHotspotClick}
+			ref={ref}
+		>
+			<div className="lfr-walkthough-hotspot-inner" />
+		</div>
+	);
+});
+
+const Walkthrough = ({closeOnClickOutside, closeable, skippable, steps}) => {
+	const [currentStepIndex, setCurrentStepIndex] = useState(0);
+
+	const memoizedTriggerReference = useMemo(() => {
+		const currentNode = steps[currentStepIndex].nodeToHighlight;
+
+		if (currentNode) {
+			return document.querySelector(currentNode);
+		}
+	}, [steps, currentStepIndex]);
+
+	const previousTrigger = usePrevious(memoizedTriggerReference);
+
+	return (
+		<>
+			<WalkthroughStep
+				closeOnClickOutside={closeOnClickOutside}
+				closeable={closeable}
+				onNext={() => {
+					const maybeNextIndex = currentStepIndex + 1;
+
+					if (steps[maybeNextIndex]) {
+						setCurrentStepIndex(maybeNextIndex);
+					}
+				}}
+				onPrevious={() => {
+					const maybePreviousIndex = currentStepIndex - 1;
+
+					if (steps[maybePreviousIndex]) {
+						setCurrentStepIndex(maybePreviousIndex);
+					}
+				}}
+				previousTrigger={previousTrigger}
+				skippable={skippable}
+				stepIndex={currentStepIndex}
+				stepLength={steps.length}
+				trigger={memoizedTriggerReference}
+				{...steps[currentStepIndex]}
+			/>
+		</>
+	);
+};
+
+/**
+ * This map humanize tuples received from dom-align
+ * library to be passed in a format that ClayPopover allows
+ */
+const ALIGNMENTS_MAP = {
+	'bottom': ['tc', 'bc'],
+	'bottom-left': ['tl', 'bl'],
+	'bottom-right': ['tr', 'br'],
+	'left': ['cr', 'cl'],
+	'left-bottom': ['br', 'bl'],
+	'left-top': ['tr', 'tl'],
+	'right': ['cl', 'cr'],
+	'right-bottom': ['bl', 'br'],
+	'right-top': ['tl', 'tr'],
+	'top': ['bc', 'tc'],
+	'top-left': ['bl', 'tl'],
+	'top-right': ['br', 'tr'],
+};
+
+Walkthrough.propTypes = {
+	closeOnClickOutside: PropTypes.bool,
+	closeable: PropTypes.bool,
+	darkbg: PropTypes.bool,
+	skippable: PropTypes.bool,
+	steps: PropTypes.arrayOf(
+		PropTypes.shape({
+			content: PropTypes.string,
+			nodeToHighlight: PropTypes.string.isRequired,
+			positioning: PropTypes.oneOf(Object.keys(ALIGNMENTS_MAP)),
```

Comment:
Instead of opening a openToast for people seeing if they passed an invalid value for the positioning, we're checking it on PropTypes and an error will be thown on Browser's console.

See:

![image](https://user-images.githubusercontent.com/7663513/167227033-eb4b1683-2bbe-4e74-863a-74bd86dc91af.png)

Where I passed `Zumbi dos Palmares` instead of one of the strings detailed as keys of ALIGNMENTS_MAP. Is this okay? 

If we decide go with openToast, should it be thrown just when NODE_ENV=development or both(in prod mode too)? 

I think since it will be run on Saas, We should consider not isolating those kind of infos when just in dev mode. Since I expect JS resources on Saas will be compiled using prod mode, right?

So, I'm inclined to don't wrap this to be shown just in development mode. 

Thoughts? @matuzalemsteles @dsanz @bryceosterhaus

Reply:
That's more than ok for me Diego